File: CPP/barn.cpp
#include <iostream>
using namespace std ;

#include "barn.h"

void Barn::render() {
	if (!this->getCreation() ){
		if (getGrass() ){
			cout << "*" ;
		}else{
			cout << "x" ;
		}
	}
	else
	{
		this->getCreation()->render() ;	
	}
	
}


int Barn::getIdentity() {
	return 0 ;
}

Barn::Barn(bool _r) {
	this->setGrass(_r) ;
}

Barn::Barn() {
	this->setGrass(false) ;
}

File: GUI-nya/Yg Package/JavaGUI/src/land/Barn.java
//#include "land.h"
package land;

public class Barn extends Land{

public char render() {
	if (this.getCreation() == null){
		if (getGrass() ){
			return '*' ; //System.out.print( "*" ) ;
		}else{
			return 'x' ; //System.out.print( "x" );
		}
	}
	else
	{
		return this.getCreation().render() ;	
	}
	
}


public int getIdentity() {
	return 0 ;
}

public Barn(boolean _r) {
	this.setGrass(_r) ;
}

public Barn() {
	this.setGrass(false) ;
}

}
